classes/socket.h

```diff
@@ -399,6 +411,19 @@ PHP_METHOD(Socket, strerror) {
 	pthreads_socket_strerror(error, return_value);
 } /* }}} */
 
+#ifdef HAVE_SHUTDOWN
+/* {{{ proto bool Socket::shutdown(int how) */
+PHP_METHOD(Socket, shutdown) {
+	zend_long how_shutdown = 0;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "l", &how_shutdown) != SUCCESS) {
```

Comment:
Is there any reason in particular why we don't default this parameter to `2`? I would assume that most of the time when we shutdown a socket, we'd like to close it for both reads and writes, so following PHP's `socket_shutdown` function seems like it would make sense.